src/Compilers/VisualBasic/Test/Emit/Emit/EditAndContinue/SymbolMatcherTests.vb

```diff
@@ -512,5 +512,327 @@ End Class
             Assert.Equal("$VB$Local_x1", mappedX1.Name)
             Assert.Null(mappedX2)
         End Sub
+
+        <Fact>
+        Public Sub TupleField_TypeChange()
+            Dim source0 = "
+Class C
+{  
+    Public x As (a As Integer, b As Integer)
+}"
+            Dim source1 = "
+Class C
+{
+    Public x As (a As Integer, b As Boolean)
+}"
+            Dim compilation0 = CreateCompilationWithMscorlib(source0, options:=TestOptions.DebugDll, references:=ValueTupleRefs)
+            Dim compilation1 = compilation0.WithSource(source1)
+
+            Dim matcher = New VisualBasicSymbolMatcher(
+                Nothing,
+                compilation1.SourceAssembly,
+                New EmitContext(),
+                compilation0.SourceAssembly,
+                New EmitContext(),
+                Nothing)
+
+            Dim member = compilation1.GetMember(Of FieldSymbol)("C.x")
+            Dim other = matcher.MapDefinition(member)
+            ' If a type changes within a tuple, we do not expect types to match.
+            Assert.Null(other)
+        End Sub
+
+        <Fact>
+        Public Sub TupleField_NameChange()
+
+            Dim source0 = "
+Class C
+{  
+    Public x As (a As Integer, b As Integer)
+}"
+            Dim source1 = "
+Class C
+{
+    Public x As (a As Integer, c As Integer)
+}"
+            Dim compilation0 = CreateCompilationWithMscorlib(source0, options:=TestOptions.DebugDll, references:=ValueTupleRefs)
+            Dim compilation1 = compilation0.WithSource(source1)
+
+            Dim matcher = New VisualBasicSymbolMatcher(
+                Nothing,
+                compilation1.SourceAssembly,
+                New EmitContext(),
+                compilation0.SourceAssembly,
+                New EmitContext(),
+                Nothing)
+
+            Dim member = compilation1.GetMember(Of FieldSymbol)("C.x")
+            Dim other = matcher.MapDefinition(member)
+            ' Types must match because just an element name was changed.
+            Assert.NotNull(other)
```

Comment:
Good idea! Thank you! Done

---
In reply to: [117564324](https://github.com/dotnet/roslyn/pull/19283#discussion_r117564324) [](ancestors = 117564324)